src/Simplify.cpp

```diff
@@ -335,6 +335,14 @@ class Simplify : public IRMutator {
             expr = mutate(Select::make(select_a->condition,
                                        select_a->true_value + select_b->true_value,
                                        select_a->false_value + select_b->false_value));
+        } else if (select_a &&
+                   is_const(b) &&
+                   (is_const(select_a->true_value) ||
+                    is_const(select_a->false_value))) {
+            // select(c, c1, c2) + c3 -> select(c, c1+c3, c2+c3)
```

Comment:
Isn't this rule potentially slower, by doing 2 adds instead of 1? I know that when I'm writing my code, I prefer the left hand side of this rule, rather than the right. I suppose LLVM should re-do this optimization, but it would be nice not to rely on that. 

I assume this rule simplifies expressions produced by bounds inference, and that's why you added it?